js/impact.js

```diff
@@ -0,0 +1,2332 @@
+/**
+ * ---------------------------------------------------------------------
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2015-2019 Teclib' and contributors.
+ *
+ * http://glpi-project.org
+ *
+ * based on GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2003-2014 by the INDEPNET Development Team.
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * GLPI is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+// Constants to represent nodes and edges
+var NODE = 1;
+var EDGE = 2;
+
+// Constants for graph direction (bitmask)
+var DEFAULT  = 0;   // 0b00
+var FORWARD  = 1;   // 0b01
+var BACKWARD = 2;   // 0b10
+var BOTH     = 3;   // 0b11
+
+// Constants for graph edition mode
+var EDITION_DEFAULT      = 1;
+var EDITION_ADD_NODE     = 2;
+var EDITION_ADD_EDGE     = 3;
+var EDITION_DELETE       = 4;
+var EDITION_ADD_COMPOUND = 5;
+
+// Constants for ID separator
+var NODE_ID_SEPERATOR = "::";
+var EDGE_ID_SEPERATOR = "->";
+
+// Constants for delta action
+var DELTA_ACTION_ADD    = 1;
+var DELTA_ACTION_UPDATE = 2;
+var DELTA_ACTION_DELETE = 3;
+
+// Load cytoscape
+var cytoscape = window.cytoscape;
+
+// Needed for JS lint validation
+var _ = window._;
+var showMenu = window.showMenu;
+
+var impact = {
```

Comment:
1. Namespaces should avec a name starting with an uppercase char (like a class name).
2. To prevent conflict with third party components, prefixing namespace with `GLPI` could be a good idea (as already done for `GLPIPlanning`.

```suggestion
var GLPIImpact = {
```